examples/platform_services/android/app/src/main/java/com/example/flutter/MainActivity.java

```diff
@@ -30,15 +30,21 @@ public void onCreate(Bundle savedInstanceState) {
           @Override
           public void onMethodCall(MethodCall call, Response response) {
             if (call.method.equals("getBatteryLevel")) {
-              getBatteryLevel(response);
+              int batteryLevel = getBatteryLevel();
+
+              if (batteryLevel != -1) {
+                response.success(batteryLevel);
+              } else {
+                response.error("UNAVAILABLE", "Battery level not available.", null);
+              }
             } else {
               throw new IllegalArgumentException("Unknown method " + call.method);
```

Comment:
I think we should handle unknown methods by `response.error` too. That's what we'll have to do on the iOS side, because of the difficulty of manual memory management in face of exceptions.